### ISSUE TITLE
Changelog should mention breaking changes

### ISSUE THREAD

**Issue by crfrolik** [typescript-eslint/typescript-eslint#2054]
**Repro**

Having `@typescript-eslint/class-name-casing` in  your `eslintrc.js` file leads to the following error when you upgrade to 3.0.0:

```
Definition for rule '@typescript-eslint/class-name-casing' was not found  @typescript-eslint/class-name-casing
```

Naturally, I looked in the changelog for info about this, and couldn't find any.
After searching around, I eventually discovered https://github.com/typescript-eslint/typescript-eslint/issues/1854

I think breaking changes are the most important thing to document in the changelog.

In addition, I got this warning, which I don't recall happening with 2.X (even though I've been using TypeScript 3.8.3 for a while):

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <3.8.0

YOUR TYPESCRIPT VERSION: 3.8.3

Please only submit bug reports when using the officially supported version.

=============
```

I suspect this is due to https://github.com/typescript-eslint/typescript-eslint/issues/1436 not having landed yet, although it was very odd that I just now got this warning. I think this should've also warranted a note in the changelog.

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `3.0.0` |
| `@typescript-eslint/parser`               | `3.0.0` |
| `TypeScript`                              | `3.8.3` |
| `ESLint`                                  | `7.0.0 |
| `node`                                    | `13.9.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0

**Comment by crfrolik** [comment]
Thanks for the link - this is helpful. I still think `CHANGELOG.md` should have notes for this stuff, *or*, if you want to only document it in one place, have a link to the Releases page. Thanks!
